### ISSUE TITLE
PointFeature.getLocation returns altitude in wrong units (Point Feature Types)

### ISSUE THREAD

**Issue by julienchastang** [Unidata/thredds#388]
[Migrated from Jira](https://bugtracking.unidata.ucar.edu/browse/TDS-672) per Sean

This issue concerns the Point Feature Types. Read in the attached nc/ncml combination (below). The data should be CF compliant to my knowledge. John looked at it too. Call the `getLocation()` method. This will return the altitude in kilometers, but is should be in meters. According to the docs: "double getAltitude() altitude in meters; missing = NaN".

[28219.476.nc](https://files.zenhub.io/56a92d81e79663ef099dfceb)

[28219.476.ncml](https://files.zenhub.io/56a92d81e79663ef099dfcea)

**Comment by cwardgar** [comment]
I can confirm this behavior on both 4.6 and 5.0. `PointFeature.getLocation()` returns an `EarthLocation`. `EarthLocation.getAltitude()` claims that it returns altitude in meters, but its implementation class–`EarthLocationImpl`–makes no attempt to convert values that aren't already in meters.

The altitude unit **is** read and made available via `DsgFeatureCollection.getAltUnits()`, but we're currently not doing a whole lot with it.

The lazy solution is to edit `EarthLocation.getAltitude()`'s Javadoc to indicate that the unit is not necessarily meters, and can be found by calling `DsgFeatureCollection.getAltUnits()`. A better solution is to store the units in each `EarthLocation`, but that's significantly more work as we'd have to change every piece of code where an instance is created.

**Comment by lesserwhirls** [comment]
I think we can force it to be in km using the udunits package. For example, we could first check if the unit is compatible with km, and if so, change it using the udunits package to get the correct scale factor and make it correct. I sort of get the feeling, especially in coordinate system classes, that we do not utilize the udunits package enough.

**Comment by cwardgar** [comment]
Yeah, but not all vertical coordinates are compatible with meters. Some are even dimensionless. It's just a bad idea to guarantee meters from that method.

**Comment by lesserwhirls** [comment]
Ok, here is what we came up with:

 * change docs to reflect reality
 * make sure we give users an idea of where to look for the units
 * All of this needs a major API review
 * Try to address this in 5.0
 * Look at OGC geotoolkit thingy and investigate for THREDDS 6.

**Comment by cwardgar** [comment]
Accomplished the first two bullets in #412. I only intend to address this in 5.0.0.
